File: ch9/9.2.2/ObjC/ModalViewSample/ModalViewSample/ViewController.m
//
//  ViewController.m
//  ModalViewSample
//
//  Created by tonyguan on 2016/10/30.
//  Copyright © 2016年 关东升. All rights reserved.
//

#import "ViewController.h"
#import "RegisterViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    CGRect screen = [[UIScreen mainScreen] bounds];
    
    self.navigationItem.title = @"登录";
    
    /// 1.添加用户ID：标签
    UILabel* labelId = [[UILabel alloc] initWithFrame:CGRectMake(80,115, 80, 21)];
    labelId.text = @"用户ID：";
    [self.view addSubview:labelId];
    
    /// 2.添加用户ID TextField
    UITextField* textFieldId = [[UITextField alloc] initWithFrame:CGRectMake(170, 106, 100, 30)];
    textFieldId.borderStyle = UITextBorderStyleRoundedRect;
    [self.view addSubview:textFieldId];
    
    /// 3.添加密码：标签
    UILabel* labelPwd = [[UILabel alloc] initWithFrame:CGRectMake(80, 178, 80, 21)];
    labelPwd.text = @"密码：";
    [self.view addSubview:labelPwd];
    
    /// 4.添加密码 TextField
    UITextField* textFieldPwd = [[UITextField alloc] initWithFrame:CGRectMake(170, 169, 100, 30)];
    textFieldPwd.borderStyle = UITextBorderStyleRoundedRect;
    [self.view addSubview:textFieldPwd];
    
    /// 5.添加 登录按钮
    UIButton* buttonLogin = [UIButton buttonWithType:UIButtonTypeSystem];
    buttonLogin.frame = CGRectMake((screen.size.width - 30)/2, 231, 50, 30);
    [buttonLogin setTitle:@"登录" forState:UIControlStateNormal];
    [self.view addSubview:buttonLogin];
    
    /// 6.添加 注册按钮
    UIButton* buttonReg = [UIButton buttonWithType:UIButtonTypeSystem];
    buttonReg.frame = CGRectMake((screen.size.width - 30)/2, 294, 50, 30);
    [buttonReg setTitle:@"注册" forState:UIControlStateNormal];
    [buttonReg addTarget:self action:@selector(onClick:) forControlEvents:UIControlEventTouchUpInside];
    
    [self.view addSubview:buttonReg];
    
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(registerCompletion:) name:@"RegisterCompletionNotification" object:nil];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    [[NSNotificationCenter defaultCenter] removeObserver:self];
}

- (void)onClick:(id)sender {
    
    RegisterViewController* registerViewController = [[RegisterViewController alloc] init];
    UINavigationController* navigationController = [[UINavigationController alloc] initWithRootViewController:registerViewController];
    
    [self presentViewController:navigationController animated:TRUE completion:nil];
}


-(void)registerCompletion:(NSNotification*)notification {
    
    NSDictionary *theData = [notification userInfo];
    NSString *username = theData[@"username"];
    
    NSLog(@"username = %@",username);
}


@end

File: ch11/11.3.3/Swift/ModalViewSample/ModalViewSample/ViewController.swift
//
//  ViewController.swift
//  ModalViewSample
//
//  Created by 关东升 on 2016-11-18.
//  本书网站：http://www.51work6.com
//  智捷课堂在线课堂：http://www.zhijieketang.com/
//  智捷课堂微信公共号：zhijieketang
//  作者微博：@tony_关东升
//  作者微信：tony关东升
//  QQ：569418560 邮箱：eorient@sina.com
//  QQ交流群：162030268
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        NotificationCenter.default.addObserver(self,
            selector: #selector(registerCompletion(_:)),
            name: NSNotification.Name(rawValue: "RegisterCompletionNotification"),
            object: nil)
        
        //监听系统通知UIApplicationDidEnterBackgroundNotification
        NotificationCenter.default.addObserver(self,
            selector: #selector(handleEnterBackground(_:)),
            name: NSNotification.Name.UIApplicationDidEnterBackground, object: nil)
        
        //监听系统通知UIApplicationWillEnterForegroundNotification
        NotificationCenter.default.addObserver(self,
            selector: #selector(handleEnterForeground(_:)),
            name: NSNotification.Name.UIApplicationWillEnterForeground, object: nil)
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        NotificationCenter.default.removeObserver(self)
    }
    
    func registerCompletion(_ notification: Notification) {
        
        let theData = notification.userInfo!
        let username = theData["username"] as! String
        
        NSLog("username = %@", username)
    }
    
    func handleEnterBackground(_ notification : Notification) {
        NSLog("进入到后台")
    }
    
    func handleEnterForeground(_ notification : Notification) {
        NSLog("回到前台")
    }
}

